src/FileSystemDirectoryHandle.ts

```diff
@@ -0,0 +1,112 @@
+import { FileSystemHandle } from './FileSystemHandle.js'
+import { FileSystemFileHandle } from './FileSystemFileHandle.js'
+import { FileSystemFolderHandleAdapter } from './interfaces.js'
+
+const kAdapter = Symbol('adapter')
+
+export class FileSystemDirectoryHandle extends FileSystemHandle implements globalThis.FileSystemDirectoryHandle {
+  /** @internal */
+  [kAdapter]: FileSystemFolderHandleAdapter
+  override readonly kind = 'directory'
+
+  constructor (adapter: FileSystemFolderHandleAdapter) {
+    super(adapter)
+    this[kAdapter] = adapter
+  }
+
+  async getDirectoryHandle (name: string, options: FileSystemGetDirectoryOptions = {}) {
+    if (name === '') throw new TypeError(`Name can't be an empty string.`)
+    if (name === '.' || name === '..' || name.includes('/')) throw new TypeError(`Name contains invalid characters.`)
+    return new FileSystemDirectoryHandle(await this[kAdapter].getDirectoryHandle(name, options))
+  }
+
+  /** @deprecated */
+  getDirectory(name: string, options: FileSystemGetDirectoryOptions = {}) {
+    return this.getDirectoryHandle(name, options)
+  }
```

Comment:
Why did u add this deprecated methods?

Reply:
I kept them for consistency, because they were defined in `@types/wicg-file-system-access`. Ideally, I should make a PR to remove them there first, but I'm not sure if they are completely unsupported now :thinking: